src/os/bluestore/KernelDevice.cc

```diff
@@ -54,8 +54,20 @@ KernelDevice::KernelDevice(CephContext* cct, aio_callback_t cb, void *cbpriv, ai
   fd_directs.resize(WRITE_LIFE_MAX, -1);
   fd_buffereds.resize(WRITE_LIFE_MAX, -1);
 
+  bool use_ioring = g_ceph_context->_conf.get_val<bool>("bluestore_ioring");
```

Comment:
might want to bypass these tests if `!defined(HAVE_LIBURING)`.

Reply:
> might want to bypass these tests if `!defined(HAVE_LIBURING)`.

That is exactly what I tried to avoid in order not to spoil the code with ifgdefs, which are ugly.

Reply:
we need to define a dummy `ioring_data` struct if `defined(HAVE_LIBURING) && defined(__x86_64__)` is false.

Reply:
> we need to define a dummy `ioring_data` struct if `defined(HAVE_LIBURING) && defined(__x86_64__)` is false.

Ha! That is exactly the reason why I did not want to keep it as a std::unique_ptr.  Will update shortly.